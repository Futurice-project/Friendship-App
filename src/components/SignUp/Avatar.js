import React from 'react';
import styled from 'styled-components/native';
import { Image, TouchableOpacity } from 'react-native';

export default class Avatar extends React.Component {
  render() {
    return (
      <TouchableOpacity
        onPress={() => this.props.updateAvatar(this.props.avatar)}
      >
        <AvatarContainer
          style={{
            backgroundColor: this.props.selected ? '#ff8a65' : '#ffffff',
          }}
        >
          <Image source={{ uri: this.props.avatar }} style={styles.avatars} />
        </AvatarContainer>
      </TouchableOpacity>
    );
  }
}

export const AvatarContainer = styled.View`
  height: 70;
  width: 70;
  background-color: #ffffff;
  border-radius: 35;
  justify-content: center;
  align-items: center;
  margin-right: 12;
`;

const styles = {
  avatars: {
    height: 60,
    width: 60,
  },
};
