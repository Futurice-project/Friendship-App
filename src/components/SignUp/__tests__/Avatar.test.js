import Avatar from '../Avatar';
import { shallow } from 'enzyme';
import React from 'react';

describe('Avatar', () => {
  test('An avatar option renders correctly', () => {
    const avatar = shallow(<Avatar />);
    expect(avatar).toMatchSnapshot();
  });
});
