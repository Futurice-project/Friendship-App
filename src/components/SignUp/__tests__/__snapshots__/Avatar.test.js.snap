// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar An avatar option renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpEmoji />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Styled(Component)
        style={
          Object {
            "backgroundColor": "#ffffff",
          }
        }
      >
        <Styled(Text) />
      </Styled(Component)>,
      "onPress": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Text) />,
        "style": Object {
          "backgroundColor": "#ffffff",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Styled(Component)
          style={
            Object {
              "backgroundColor": "#ffffff",
            }
          }
        >
          <Styled(Text) />
        </Styled(Component)>,
        "onPress": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Text) />,
          "style": Object {
            "backgroundColor": "#ffffff",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
